admission-webhook/src/utils/validation-server.ts

```diff
@@ -0,0 +1,250 @@
+import fastify from 'fastify';
+import pino from 'pino';
+import path from 'path';
+import {readFileSync} from 'fs';
+import {Message, Resource, RuleLevel} from '@monokle/validation';
+import {V1ObjectMeta} from '@kubernetes/client-node';
+import {ValidatorManager} from './validator-manager.js';
+
+export type ValidationServerOptions = {
+  port: number
+  host: string
+};
+
+export type AdmissionRequestObject = {
+  apiVersion: string
+  kind: string
+  metadata: V1ObjectMeta
+  spec: any
+  status: any
+};
+
+export type AdmissionRequest = {
+  apiVersion: string
+  kind: string
+  request: {
+    name: string
+    namespace: string
+    uid: string
+    object: AdmissionRequestObject
+  }
+};
+
+// See
+// https://pkg.go.dev/k8s.io/api/admission/v1#AdmissionResponse
+// https://kubernetes.io/docs/reference/config-api/apiserver-admission.v1/#admission-k8s-io-v1-AdmissionReview
+export type AdmissionResponse = {
+  kind: string
+  apiVersion: string
+  response: {
+      uid: string
+      allowed: boolean
+      warnings?: string[]
+      status: {
+          message: string
+      }
+  }
+};
+
+export type Violation = {
+  ruleId: string
+  message: Message
+  level?: RuleLevel
+  actions: string[]
+}
+
+export class ValidationServer {
+  private _server: ReturnType<typeof fastify>;
+  private _shouldValidate: boolean
+
+  constructor(
+    private readonly _validators: ValidatorManager,
+    private readonly _ignoredNamespaces: string[],
+    private readonly _logger: ReturnType<typeof pino>,
+    private readonly _options: ValidationServerOptions = {
+      port: 8443,
+      host: '0.0.0.0'
+    }
+  ) {
+    this._shouldValidate = false;
+
+    this._server = fastify({
+      https: {
+        key: readFileSync(path.join('/run/secrets/tls', 'tls.key')),
+        cert: readFileSync(path.join('/run/secrets/tls', 'tls.crt'))
+      }
+    });
+
+    this._initRouting();
```

Comment:
```suggestion
    void this._initRouting();
```

Tooling likes to complain when you do not handle promises. It's best to give an explicit hint that you do not care about it. Alternative is to `.catch(noop)` to ignore the error. In this case Fastify never throws so should be fine - I even think the `async` can be removed all together from initRouting.

Reply:
Looks like leftover, this function is sync in fact. Fixed 👍